com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -297,34 +197,33 @@ private void FixedUpdate()
                 return;
             }
 
-            if (CanUpdateTransform())
+            if (IsNetworkStateDirty)
             {
-                m_NetworkPosition.Value = m_CurrentPosition;
-                m_NetworkRotation.Value = m_CurrentRotation;
-                m_NetworkWorldScale.Value = m_CurrentScale;
+                if (CanUpdateTransform)
+                {
+                    UpdateNetworkState();
+                }
+                else
+                {
+                    ApplyNetworkState(m_NetworkState.Value);
+                }
             }
-            else if (m_CurrentPosition != m_OldPosition ||
-                m_CurrentRotation != m_OldRotation ||
-                m_CurrentScale != m_OldScale
-            )
-            {
-                Debug.LogError($"Trying to update transform's position for object {gameObject.name} with ID {NetworkObjectId} when you're not allowed, please validate your {nameof(NetworkTransform)}'s authority settings. It's also possible you have other systems updating your transform's position, such as a rigid body for example.", gameObject);
-                m_CurrentPosition = m_NetworkPosition.Value;
-                m_CurrentRotation = m_NetworkRotation.Value;
-                m_CurrentScale = m_NetworkWorldScale.Value;
+        }
 
-                m_OldPosition = m_CurrentPosition;
-                m_OldRotation = m_CurrentRotation;
-                m_OldScale = m_CurrentScale;
-            }
+        /// <summary>
+        /// Updates the NetworkTransform's authority model at runtime
+        /// </summary>
+        public void SetAuthority(NetworkAuthority authority)
```

Comment:
```suggestion
        internal void SetAuthority(NetworkAuthority authority)
```
I think we should keep this internal for now as it is easy to mis-use and does not work as you would expect.